### PR TITLE
Added required features

### PR DESCRIPTION
**Pull request by arghyadipchak**
- [x] Parsing of collections
- [X] Execution of parsed scripts
- [X] Selection of scripts
- [X] Templating in scripts
- [X] Tests

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -26,22 +32,74 @@ pub enum OS {
...
+impl OS {
+    /**
+    Returns [`OS`] respective to current system
 
-    match os {
-        OS::MacOs => {
-            filename.push_str("macos.yaml");
-        }
-        OS::Windows => filename.push_str("windows.yaml"),
-        OS::Linux => {
```
Comment (sn99) [inline src/lib.rs:38]: Make it an `error` not `panic!`

**Review by sn99 — CHANGES_REQUESTED** [review]

**Comment by sn99** [comment]
I will review next in the upcoming days

File: Cargo.toml [hunk 2]
```diff
@@ -2,15 +2,29 @@
...
+serde_yaml = "0.9.21"
+toml = "0.7.4"
+
+[dev-dependencies]
+glob = "0.3.1"
+
+[profile.release]
+codegen-units = 1
+lto = true
+panic = "abort"
```
[the remark this answers is not in the document]
Comment (arghyadipchak) [reply Cargo.toml]: Done

File: src/lib.rs [hunk 3]
```diff
@@ -26,22 +32,74 @@ pub enum OS {
...
+# Panics
+
+Panics for [`OS::Linux`]
+*/
+pub fn get_collection(os: OS) -> Result<CollectionData, Box<dyn std::error::Error>> {
+    let mut coll_file = PathBuf::from("collections");
+
+    coll_file.push(match os {
+        OS::MacOs => "macos.yaml",
+        OS::Linux => panic!("No rules yet!"),
```
[the remark this answers is not in the document]
Comment (arghyadipchak) [reply src/lib.rs]: Done 👍🏼

**Pull request merged**